download/download.go

```diff
@@ -247,25 +244,23 @@ func (d *Downloader) oneShot(ctx context.Context) (bool, error) {
 		if d.f != nil {
 			d.f(ctx, Update{ETag: "", Bundle: nil, Error: err, Metrics: m, Raw: nil})
 		}
-
-		return false, err
+		return err
 	}
 
 	d.etag = resp.etag
+	d.longPollingEnabled = resp.longPoll
 
 	if d.f != nil {
 		d.f(ctx, Update{ETag: resp.etag, Bundle: resp.b, Error: nil, Metrics: m, Raw: resp.raw})
 	}
-
-	return resp.longPoll, nil
+	return nil
 }
 
 func (d *Downloader) download(ctx context.Context, m metrics.Metrics) (*downloaderResponse, error) {
 	d.logger.Debug("Download starting.")
 
 	d.client = d.client.WithHeader("If-None-Match", d.etag)
-
-	if d.config.Polling.LongPollingTimeoutSeconds != nil {
+	if d.longPollingEnabled && d.config.Polling.LongPollingTimeoutSeconds != nil {
```

Comment:
Nit: Just `if d.longPollingEnabled` check should be good, correct ?

Reply:
it's depend:

if ` d.config.Polling.LongPollingTimeoutSeconds` has nil value, (long polling is not configured) but bundle server send `Content-Type: application/vnd.openpolicyagent.bundles` long polling will be enabled, so next time download is invoked, a `panic: runtime error: invalid memory address or nil pointer dereference` occured when the prefere will be configured/added to http client. 

We some possibility for avoid this segfault:
1. keep like that 
2.  add/restore the if in oneShot method for take in consideration resp.longPoll only if opa has long polling configured.
3. change `isLongPollSupported`: this method should return `false`  if long polling is not configured
4. has default value, but i don't see a good default value.
5. Other ?

Reply:
@srenatus what we decide for this case ? 